### PR TITLE
[Windows][Unicode] wstring to string in windows failed

### PR DESCRIPTION
**Pull request by zhaixuejun1993**
### Details:
 - Windows std::wstring convert to std::string some unicode will be ???, such as L"unicode_Яㅎあ" -> "unicode_???"

### Tickets:
 - CVS-141444
 - CVS-141729

### CODE REVIEW

File: src/common/util/src/file_util.cpp [hunk 1]
```diff
@@ -341,9 +341,9 @@ void ov::util::convert_path_win_style(std::string& path) {
 
 std::string ov::util::wstring_to_string(const std::wstring& wstr) {
 #    ifdef _WIN32
-    int size_needed = WideCharToMultiByte(CP_ACP, 0, &wstr[0], (int)wstr.size(), NULL, 0, NULL, NULL);
+    int size_needed = WideCharToMultiByte(CP_UTF8, 0, &wstr[0], (int)wstr.size(), NULL, 0, NULL, NULL);
```
Comment (ilya-lavrenov) [inline src/common/util/src/file_util.cpp:344]: looks like it reverts this commit https://github.com/openvinotoolkit/openvino/commit/976298c41725ef7618e06a260b3f50f38e7d7a35

Comment (zhaixuejun1993) [reply src/common/util/src/file_util.cpp:350]: @ilya-lavrenov  Another solution for catch dir, had synced with river. CP_ACP -> CP_UTF8 will be better.

File: src/common/util/include/openvino/util/file_util.hpp [hunk 2]
```diff
@@ -125,6 +125,15 @@ bool is_absolute_file_path(const std::string& path);
  */
 void create_directory_recursive(const std::string& path);
 
+#ifdef OPENVINO_ENABLE_UNICODE_PATH_SUPPORT
+/**
+ * @brief Interface function to create directorty recursively by given path
+ * @param path - path to file wide-string, can be relative to current working directory
+ * @throw runtime_error if any error occurred
+ */
+void create_directory_recursive(const std::wstring& path);
```
Comment (riverlijunjie) [inline src/common/util/include/openvino/util/file_util.hpp:134]:
```suggestion
void create_directory_recursive_w(const std::wstring& path);
```

File: src/common/util/src/file_util.cpp [hunk 3]
```diff
@@ -416,6 +422,27 @@ void ov::util::create_directory_recursive(const std::string& path) {
     }
 }
 
+#ifdef OPENVINO_ENABLE_UNICODE_PATH_SUPPORT
+void ov::util::create_directory_recursive(const std::wstring& path) {
+    if (path.empty() || directory_exists(path)) {
+        return;
+    }
+
+    std::size_t pos = path.rfind(ov::util::FileTraits<char>::file_separator);
```
Comment (riverlijunjie) [inline src/common/util/src/file_util.cpp:431]: Should it be wstring?

File: src/inference/src/cache_manager.hpp [hunk 4]
```diff
@@ -118,27 +118,44 @@ class FileStorageCacheManager final : public ICacheManager {
 
 private:
     void write_cache_entry(const std::string& id, StreamWriter writer) override {
-        // Fix the bug caused by pugixml, which may return unexpected results if the locale is different from "C".
-        ScopedLocale plocal_C(LC_ALL, "C");
```
Comment (riverlijunjie) [inline src/inference/src/cache_manager.hpp:122]: It can avoid Locale issue?

File: src/common/util/include/openvino/util/file_util.hpp (hunk 2 above)
Comment (zhaixuejun1993) [reply src/common/util/include/openvino/util/file_util.hpp:134]: why here need the different name for wstring? The other api also using the same name overload

File: src/common/util/src/file_util.cpp (hunk 3 above)
Comment (zhaixuejun1993) [reply src/common/util/src/file_util.cpp:431]: you mean the "file_separator" should be wstring?

File: src/inference/src/cache_manager.hpp (hunk 4 above)
Comment (zhaixuejun1993) [reply src/inference/src/cache_manager.hpp:122]: recovered

**Review by riverlijunjie — APPROVED** [review]
LGTM

**Review by jane-intel — APPROVED** [review]
Looks good to me

**Review by olpipi — APPROVED** [review]
I think we need to add tests for Unicode cases

**Pull request merged**